add.go

```diff
@@ -87,7 +88,7 @@ func (b *Builder) Add(destination string, extract bool, source ...string) error
 				return fmt.Errorf("error ensuring directory %q exists: %v)", dest, err)
 			}
 			logrus.Debugf("copying %q to %q", src+string(os.PathSeparator)+"*", d+string(os.PathSeparator)+"*")
-			if err := archive.CopyWithTar(src, d); err != nil {
+			if err := chrootarchive.CopyWithTar(src, d); err != nil {
 				return fmt.Errorf("error copying %q to %q: %v", src, d, err)
 			}
 			continue
```

Comment:
can/should the !archive.IsArchivePath on line 95 change to chrootarchive?  I think if you remove that you can remove the include for archive too.